src/groups/heisenberg.jl

```diff
@@ -0,0 +1,255 @@
+
+@doc raw"""
+    HeisenbergGroup{n} <: AbstractDecoratorManifold{ℝ}
+
+Heisenberg group `HeisenbergGroup(n)` is the group of ``(n+2) × (n+2)`` matrices
+
+```math
+\begin{bmatrix} 1 & \mathbf{a} & c \\
+\mathbf{0} & I_n & \mathbf{b} \\
+0 & \mathbf{0} & 1 \end{bmatrix}
+```
+
+where ``I_n`` is the ``n×n`` unit matrix, ``\mathbf{a}`` is a row vector of length ``n``,
+``\mathbf{b}`` is a column vector of length ``n`` and ``c`` is a real number.
+The group operation is matrix multiplication.
+
+The left-invariant metric on the manifold is used.
+"""
+struct HeisenbergGroup{n} <: AbstractDecoratorManifold{ℝ} end
+
+function active_traits(f, ::HeisenbergGroup, args...)
+    return merge_traits(
+        IsGroupManifold(MultiplicationOperation()),
+        IsEmbeddedManifold(),
+        HasLeftInvariantMetric(),
+    )
+end
+
+function _heisenberg_a_view(::HeisenbergGroup{n}, p) where {n}
```

Comment:
I do not like this name much, do we maybe have a more semantic name? It is an internal function so maybe this is als not so urgent.

Reply:
It's a function that makes a view to the "a" part of Heisenberg point/vector matrix. "a" part as in the notation used in docstring. I can change that if you have a better name.